### PR TITLE
Improve cpplint_wrapper.py, and add unit testing

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
A portion of #2539.

This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/2932)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
@rpoyner-tri for feature review, please.

**Comment by rpoyner-tri** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-:-KNZcNY1DZQhlLTd3a-7:1417088721)*

**Comment by jwnimmer-tri** [comment]
@david-german-tri for platform review, please... or if python isn't your thing, feel free to bump to another; ggould-tri handled the prior PR in this series.

**Comment by david-german-tri** [comment]
Yeah, I don't think I can review this productively.  +@ggould-tri, -@david-german-tri 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-:-KNbOL59ZAFNy0bJR41Q:-1284899851)*

**Comment by jwnimmer-tri** [comment]
The `drake-ci` scripts have evolved since the last time Jenkins was run.  I want to make sure the build system pieces still work, so @drake-jenkins-bot retest this please.

**Comment by ggould-tri** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/common/test/cpplint_wrapper.py, line 9 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-KNbOvtystdBIS5wbd_1:-KNbOvtystdBIS5wbd_2:-1639373288) ([raw file](https://github.com/robotlocomotion/drake/blob/bf2565b2a0ff720bf54c0b11975b1bd288e7e65e/drake/common/test/cpplint_wrapper.py#L9)):*
> ```Python
> 
> import argparse
> import functools
> ```

I smile with joy at every mention of `functools`.

---

*[drake/common/test/cpplint_wrapper_test.py, line 15 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-KNbQ7lV4ow5tBChE3RT:-KNbQ7lV4ow5tBChE3RU:116124284) ([raw file](https://github.com/robotlocomotion/drake/blob/bf2565b2a0ff720bf54c0b11975b1bd288e7e65e/drake/common/test/cpplint_wrapper_test.py#L15)):*
> ```Python
> 
>     def get_relative_filename(self, filename):
>         """Return an absolute filename, relative to this test program."""
> ```

This function name and docstring make my head hurt.  If `get_relative`... is documented to return an _absolute_ filename, which is `relative to this` then I have no idea what is going on.

Consider an example, perhaps?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932)*

**Comment by jwnimmer-tri** [comment]
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/common/test/cpplint_wrapper_test.py, line 15 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-KNbQ7lV4ow5tBChE3RT:-KNbct_9vWHlEhOZATvD:-859569042) ([raw file](https://github.com/robotlocomotion/drake/blob/bf2565b2a0ff720bf54c0b11975b1bd288e7e65e/drake/common/test/cpplint_wrapper_test.py#L15)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary>
> This function name and docstring make my head hurt.  If `get_relative`... is documented to return an _absolute_ filename, which is `relative to this` then I have no idea what is going on.
> 
> Consider an example, perhaps?
</details>
Done (renamed function and its parameter).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932)*

**Comment by ggould-tri** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-:-KNbgXRCTeT9CldNwDFk:1417088721)*

**Comment by jwnimmer-tri** [comment]
Of course jenkins is failing because WITH_GOOGLE_STYLEGUIDE is now turned off everywhere.  I'll need to think for a moment about where and how to turn it on.

**Comment by jamiesnape** [comment]
cpplint builds do not run the superbuild in any case.

**Comment by jamiesnape** [comment]
But if you have a regression test, we can switch it on. You should guard the test appropriately here for when cpplint is not found.

**Comment by jamiesnape** [comment]
@drake-jenkins-bot retest this please

File: drake/common/test/CMakeLists.txt [hunk 1]
```diff
@@ -111,3 +111,9 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
     drake_deprecated_test_nocompile
     PROPERTIES WILL_FAIL TRUE)
 endif()
+
+# Rules for cpplint_wrapper.py.
+if(PYTHONINTERP_FOUND)
```
Comment (jamiesnape) [inline drake/common/test/CMakeLists.txt]: `AND WITH_GOOGLE_STYLEGUIDE`.

**Comment by jamiesnape** [comment]
> cpplint builds do not run the superbuild in any case.

We can update the ci script to use the superbuild cpplint once #2896 lands.

**Comment by jwnimmer-tri** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/common/test/CMakeLists.txt, line 116 \[r2\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-KNZP_zF2mxO9pOzcGI6-r2-116:-KNceER5h-j0HzD6Uphy:-867827955) ([raw file](https://github.com/robotlocomotion/drake/blob/4256744144891243ddfb3d83ca1e0395085d6d25/drake/common/test/CMakeLists.txt#L116)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary>
> `AND WITH_GOOGLE_STYLEGUIDE`.
</details>
By what mechanism should that variable be available to this `CMakeLists.txt`?  I see it in `CMakeCache.txt` at the superbuild level, but not at the drake-build level, and adding the extra condition here turns off the test even with I do have `WITH_GOOGLE_STYLEGUIDE` enabled in the superbuild.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932)*

**Comment by jamiesnape** [comment]
Add `-DWITH_GOOGLE_STYLEGUIDE:BOOL=${WITH_GOOGLE_STYLEGUIDE}` here:
https://github.com/RobotLocomotion/drake/blob/master/CMakeLists.txt#L154

**Comment by jwnimmer-tri** [comment]
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/common/test/CMakeLists.txt, line 116 \[r2\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-KNZP_zF2mxO9pOzcGI6-r2-116:-KNd__1rh5kwxfPrGsEl:-499146009) ([raw file](https://github.com/robotlocomotion/drake/blob/4256744144891243ddfb3d83ca1e0395085d6d25/drake/common/test/CMakeLists.txt#L116)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> By what mechanism should that variable be available to this `CMakeLists.txt`?  I see it in `CMakeCache.txt` at the superbuild level, but not at the drake-build level, and adding the extra condition here turns off the test even with I do have `WITH_GOOGLE_STYLEGUIDE` enabled in the superbuild.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932)*

**Comment by jwnimmer-tri** [comment]
Jenkins has passed.  I will still wait for someone (@ggould-tri or @rpoyner-tri) to review the r4 changes before I merge.

**Comment by rpoyner-tri** [comment]
:lgtm:, yet again.

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2932#-:-KNdwnYl2RYBt9jV2-YQ:2085509835)*

**Pull request merged**